packages/component-submission/server/aggregates/Submission.test.js

```diff
@@ -192,33 +176,287 @@ describe('Submission', () => {
       )
       jest
         .spyOn(Manuscript, 'find')
-        .mockReturnValue(createMockObject({}, mockManuscriptSave))
-      jest.spyOn(File, 'findByManuscriptId')
+        .mockReturnValue(
+          createMockObject({ status: 'INITIAL' }, mockManuscriptSave),
+        )
 
-      const submission = await new Submission({
-        models: { Manuscript, File },
-        services: {},
-      }).initialize()
+      const submission = await createSubmission().initialize()
 
-      submission.updateManuscript({})
+      await submission.updateManuscript({})
       expect(mockManuscriptSave).toBeCalled()
     })
 
-    it('pases the correct parameters to the mergeWith function', async () => {
+    it('passes the correct parameters to the mergeWith function', async () => {
       const mockMergeFunction = jest.fn(manuscript => manuscript)
       utils.mergeObjects.mockImplementationOnce(mockMergeFunction)
       const mockManuscriptSave = jest.fn()
-      const manuscriptMock = createMockObject({}, mockManuscriptSave)
+      const manuscriptMock = createMockObject(
+        { status: 'INITIAL' },
+        mockManuscriptSave,
+      )
       jest.spyOn(Manuscript, 'find').mockReturnValue(manuscriptMock)
-      jest.spyOn(File, 'findByManuscriptId')
 
-      const submission = await new Submission({
-        models: { Manuscript, File },
-        services: {},
-      }).initialize()
+      const submission = await createSubmission().initialize()
 
-      submission.updateManuscript({})
+      await submission.updateManuscript({})
       expect(mockMergeFunction).toBeCalledWith(manuscriptMock, {})
     })
   })
+
+  describe('updateAuthorTeam', () => {
+    it('should update the author', async () => {
+      const submission = await createSubmission().initialize()
+      submission.updateAuthorTeam(1)
+
+      expect(submission.teams[0]).toEqual({
+        objectType: 'manuscript',
+        role: 'author',
+        teamMembers: [{ alias: 1, meta: { corresponding: true } }],
+      })
+    })
+
+    it('should update the existing author', async () => {
+      const submission = await createSubmission().initialize()
+      submission.updateAuthorTeam(1)
+      expect(submission.teams[0]).toEqual({
+        objectType: 'manuscript',
+        role: 'author',
+        teamMembers: [{ alias: 1, meta: { corresponding: true } }],
+      })
+      submission.updateAuthorTeam(2)
+      expect(submission.teams[0]).toEqual({
+        objectType: 'manuscript',
+        role: 'author',
+        teamMembers: [{ alias: 2, meta: { corresponding: true } }],
+      })
+    })
+  })
+
+  describe('updateEditorTeams', () => {
```

Comment:
Can we add a test for the validation check here https://github.com/elifesciences/elife-xpub/pull/2009/files#diff-5b518139d4766e3a75dc9c2cb1213c90R122